third_party/txt/src/txt/paragraph_txt.cc

```diff
@@ -1000,23 +1000,25 @@ void ParagraphTxt::Layout(double width) {
         line_metrics.run_metrics.emplace(run_key, &run.style());
         font.getMetrics(&line_metrics.run_metrics.at(run_key).GetFontMetrics());
 
-        SkFontMetrics* metrics;
-        metrics = &line_metrics.run_metrics.at(run_key).GetFontMetrics();
         Range<double> record_x_pos(
             glyph_positions.front().x_pos.start - run_x_offset,
             glyph_positions.back().x_pos.end - run_x_offset);
         if (run.is_placeholder_run()) {
           paint_records.emplace_back(
               run.style(), SkPoint::Make(run_x_offset + justify_x_offset, 0),
-              builder.make(), *metrics, line_number, record_x_pos.start,
+              builder.make(),
```

Comment:
Can we have a test that exercises this?  Ideally one that causes a segfault without it...

Reply:
This change isn't actually what caused the segfault. It was the change in line 1767 that fixes it. Here, I figured I would just use the metrics value directly instead of caching it and messing with pointers as it caused me pain when debugging.

The only case that I know of that repros this is a quite convoluted case in g3 that segfaults, and it would be quite difficult to extract out as it is a generated example out of a bunch of protos and assets. I've tried a manually coded version of it based on what it would generate, but I could not get it to segfault.

Reply:
I understand that now from offline - should we rever this part then?

Reply:
Sure! I can make this a bit cleaner anyways.